pkg/cmd/adm/register_member_test.go

```diff
@@ -178,11 +86,20 @@ func TestRegisterMember(t *testing.T) {
 		}
 
 		// when
-		err := registerMemberCluster(ctx, addClusterCommand, 1*time.Second, newRegisterMemberArgsWith(hostKubeconfig, memberKubeconfig, false))
+		err := registerMemberCluster(ctx, newRegisterMemberArgsWith(hostKubeconfig, memberKubeconfig, false))
 
 		// then
 		require.NoError(t, err)
-		assert.Equal(t, 2, *counter)
+		// check the expected secrets are there with the kubeconfigs
+		// the member kubeconfig secret in the host namespace
+		verifyToolchainClusterSecret(t, fakeClient, "toolchain-host-operator", "toolchain-member-operator", memberToolchainClusterName)
+		// the host secret in the member namespace
+		verifyToolchainClusterSecret(t, fakeClient, "toolchain-member-operator", "toolchain-host-operator", hostToolchainClusterName)
+		tcs := &toolchainv1alpha1.ToolchainClusterList{}
+		require.NoError(t, fakeClient.List(context.TODO(), tcs, runtimeclient.InNamespace("toolchain-host-operator")))
+		assert.Len(t, tcs.Items, 1)
+		require.NoError(t, fakeClient.List(context.TODO(), tcs, runtimeclient.InNamespace("toolchain-member-operator")))
+		assert.Len(t, tcs.Items, 1)
```

Comment:
why listing them? if you want to check that there were not more than one TC created, then fine, but let's verify the name of the TC:
```suggestion
		tcs := &toolchainv1alpha1.ToolchainClusterList{}
		require.NoError(t, fakeClient.List(context.TODO(), tcs, runtimeclient.InNamespace("toolchain-host-operator")))
		assert.Len(t, tcs.Items, 1)
		assert.Equal(t, memberToolchainClusterName, tcs.Items[0].Name)
		require.NoError(t, fakeClient.List(context.TODO(), tcs, runtimeclient.InNamespace("toolchain-member-operator")))
		assert.Len(t, tcs.Items, 1)
		assert.Equal(t, hostToolchainClusterName, tcs.Items[0].Name)
```

Reply:
one more thing, I understand that you were not motivated in verifying the content of the ToolchainCluster CRs as this will go away anyway, but let's at least verify that the linked secret matches the expected one.

Reply:
yeah I want to check also that it doesn't create more than one TC. Thanks for your suggestion I've applied it in https://github.com/kubesaw/ksctl/pull/53/commits/6be692c5f65a23b3c59439e767331fc20288dbe5